packages/gui/src/components/nfts/gallery/NFTGallery.tsx

```diff
@@ -117,70 +211,257 @@ export default function NFTGallery() {
+                  control={<Checkbox />}
+                  label={t`${key} (${nftTypes[key]})`}
+                  checked={typeFilter.indexOf(key) === -1}
+                  onClick={(e) => {
+                    e.preventDefault();
+                    e.stopPropagation();
+                    setTypeFilter(
+                      typeFilter.indexOf(key) === -1 ? typeFilter.concat(key) : typeFilter.filter((x) => key !== x)
+                    );
+                  }}
+                />
+              ))}
+          </RadioGroup>
+        </FormControl>
+      </div>
+    );
+  }
+
+  function toggleVisibilityFilter(e, type: string) {
+    e.preventDefault();
+    e.stopPropagation();
+    setVisibilityFilters(
+      visibilityFilters.indexOf(type) === -1
+        ? visibilityFilters.concat(type)
+        : visibilityFilters.filter((x) => x !== type)
+    );
+  }
+
+  function checkedNftTypes(nftTypes: any, typeFilter: string[]) {
+    return Object.keys(nftTypes).filter((key) => typeFilter.indexOf(key) > -1).length;
+  }
+
   return (
     <LayoutDashboardSub
       // sidebar={<NFTGallerySidebar onWalletChange={setWalletId} />}
       header={
-        <Flex gap={2} alignItems="center" flexWrap="wrap" justifyContent="space-between">
-          <NFTProfileDropdown onChange={setWalletId} walletId={walletId} />
-          <Flex justifyContent="flex-end" alignItems="center">
-            {/*
-            <Search
-              onChange={setSearch}
-              value={search}
-              placeholder={t(defineMessage({ message: `Search...` }))}
-            />
-            */}
-            {/*
-            <NFTContextualActions selection={selection} />
-            */}
-            <Box width={{ xs: 300, sm: 330, md: 600, lg: 780 }}>
-              <Flex gap={1}>
-                <WalletReceiveAddressField variant="outlined" size="small" fullWidth />
-                <DropdownActions
-                  label={<Trans>Filters</Trans>}
-                  startIcon={<FilterListIcon />}
-                  endIcon={undefined}
-                  variant="text"
-                  color="secondary"
-                  size="large"
+        <>
+          <Flex gap={2} alignItems="stretch" flexWrap="wrap" justifyContent="space-between">
+            <NFTProfileDropdown onChange={setWalletId} walletId={walletId} />
+            <Search onChange={setSearch} placeholder={t`Search...`} />
+            <WalletReceiveAddressField variant="outlined" size="small" fullWidth />
+            <MultiSelectAndFilterWrapper className={inMultipleSelectionMode ? 'active' : ''}>
+              <MultiSelectIcon onClick={() => toggleMultipleSelection(!inMultipleSelectionMode)} />
+              <FilterIcon />
+            </MultiSelectAndFilterWrapper>
+          </Flex>
+
+          <Flex gap={2} alignItems="center" flexWrap="wrap" justifyContent="space-between" sx={{ padding: '10px 0' }}>
+            {t`Showing ${showCount()} of ${allowNFTsFiltered.length} items`}
```

Comment:
If I switch to a wallet without any NFTs, the total count doesn't show properly. In the screenshot below, I switched from my "All NFTs" view to "Profile 1" (which doesn't have any NFTs).

![image](https://user-images.githubusercontent.com/339312/207398574-66888afd-4ad7-4720-b7be-a0c0eaa5fe2f.png)

Reply:
Tracked as a new issue: https://github.com/Chia-Network/chia-blockchain-gui/issues/1250